src/crypto/tls/common.go

```diff
@@ -747,6 +729,11 @@ type Config struct {
 	// payload.
 	ServerECHProvider ECHProvider
 
+	// EventHandler, if set, is called by the client and server at various
+	// points during the handshake to handle specific events. This callback can
+	// be used to record interesting metrics.
```

Comment:
We should say something about the, uh, stability of these events? For example:

```suggestion
	// be used to record interesting metrics. Callers MUST NOT rely on the presence or order of 
	// events generated by this callback, as they may change at any time.
```

In a former life, clients ossified on such functionality and caused *all sorts of damage*.

Reply:
- Added a note about this API being experimental.
- Prefixed `Event` and `Config.EventHandler` with `EXP_` in order to signal callers that they shouldn't rely on this API yet.